### DISCUSSION TITLE
Add custom column to 'GraphQL' select query

### DISCUSSION THREAD

**Discussion by JVariance** [supabase/supabase#2624] [Questions] [answered]
Can I somehow add a column that doesn't exist on the queried table to a 'GraphQL'-style select?

Like:
```javascript
await supabase
.from('table')
.select(`
  id,
  ...,
  true as exampleBool,
  42 as exampleNum
`);

//Result:
table = {
  id: 1,
  ...,
  exampleBool: true,
  exampleNum: 42
}
```

**Comment by silentworks** [comment]
This is not possible, why not just do this in javascript land rather then trying to get it back as part of the results?

**Reply by JVariance** [reply]
I somehow liked that idea more than looping through the array in js.
But thanks ✌️

**Comment by lawrencecchen** [comment]
Another solution is to create a postgres view and query that instead. Not particularly relevant for simple cases, but helpful if you want joins, more complicated queries, or to offload work to the database.
https://postgrest.org/en/v7.0.0/api.html
<img width="726" alt="Screen Shot 2021-07-29 at 7 03 15 PM" src="https://user-images.githubusercontent.com/54008264/127588778-eafdccde-efd5-49da-a55c-369cce21df5c.png">

**Reply by JVariance** [reply]
Thanks, I had Views in mind, but I tried to avoid writing complex joins through nested/foreign keyed tables.

**Comment by steve-chavez** [comment, marked as answer]
[Computed columns](https://postgrest.org/en/latest/api.html#computed-columns) look fit for this case. Though they usually "compute" a value based on other columns they can also be used for constants.

```sql
CREATE FUNCTION "exampleNum"(my_table) RETURNS int AS $$
  SELECT 42;
$$ LANGUAGE SQL;
```

```js
await supabase
.from('my_table')
.select(`
  id,
  ...,
  exampleNum
`);

//Result:
table = {
  id: 1,
  ...,
  exampleNum: 42
}

```
